src/query/expression/src/evaluator.rs

```diff
@@ -951,13 +951,96 @@ impl<'a> Evaluator<'a> {
         unreachable!("expr is not a set returning function: {expr}")
     }
 
+    fn run_array_fold(&self, column: &Column, expr: &Expr) -> Result<Value<AnyType>> {
```

Comment:
Return `Result<Scalar>` here, so that we don't need to convert the value back to scalar in `run_lambda`